src/runDiff.ts

```diff
@@ -0,0 +1,225 @@
+  /** Direct link to the job execution */
+  html_url: string;
+  /** ID of the executed job */
+  job_id: string;
+  /** 
+   * Result code indicating the outcome:
+   * - '0': Success with no changes
+   * - '1': Error occurred
+   * - '2': Success with changes detected
+   */
+  result_code: "0" | "1" | "2";
+  /** Name of the CDKTF stack used */
+  stack: string;
+  /** Human-readable summary of changes */
+  summary: string;
+}
+
+/**
+ * Executes CDKTF diff command, parses the output, and provides structured results.
+ * Handles different output patterns to determine if there are changes, errors, or no changes.
+ */
+export class RunDiff {
+  octokit: Octokit & Api & { paginate: PaginateInterface };
+  context: Context;
+  inputs: ActionInputs;
+
+  /**
+   * Initialize clients and member variables.
+   */
+  constructor() {
+    // Get inputs first
+    this.inputs = {
+      githubToken: core.getInput("github_token", { required: true }),
+      jobName: core.getInput("job_name", { required: true }),
+      outputFilename: core.getInput("output_filename", { required: true }),
+      ref: core.getInput("ref", { required: true }),
+      stack: core.getInput("stack", { required: true }),
+      stubOutputFile: core.getInput("stub_output_file"),
+      terraformVersion: core.getInput("terraform_version") || "1.8.0",
+      workingDirectory: core.getInput("working_directory") || "./",
+      skipSynth: core.getBooleanInput("skip_synth"),
+      artifactName: core.getInput("artifact_name"),
+    };
+
+    // Then create octokit with the token
+    this.octokit = github.getOctokit(this.inputs.githubToken);
+    this.context = github.context;
+    core.debug("Context:");
+    core.debug(JSON.stringify(this.context));
+  }
+
+  /**
+   * Main execution method. Gets the job ID, runs the CDKTF diff, and processes the results.
+   * Sets action outputs and writes results to a file. Fails the action if errors are detected.
+   */
+  async run() {
+    // Get job information
+    const { job_id, html_url } = await this.getJobId();
+
+    // Run the diff
+    const { result_code, summary } = await this.runDiff();
+
+    // Prepare outputs
+    const outputs: ActionOutputs = {
+      html_url,
+      job_id: job_id.toString(),
+      result_code,
+      stack: this.inputs.stack,
+      summary
+    };
+
+    // Set action outputs
+    Object.entries(outputs).forEach(([key, value]) => {
+      core.setOutput(key, value);
+    });
+
+    // Write outputs to file
+    const outputPath = path.join(this.inputs.workingDirectory, this.inputs.outputFilename);
+    fs.writeFileSync(outputPath, JSON.stringify(outputs));
+
+    if(result_code === "1") {
+      core.setFailed(summary);
+    }
+  }
+
+  /**
+   * Retrieves the job ID and HTML URL for the current workflow job.
+   * 
+   * @returns Promise containing the job ID and HTML URL
+   * @throws Error if the job cannot be found
+   */
+  async getJobId(): Promise<{ job_id: number; html_url: string }> {
```

Comment:
With the double return, this is quite a deceptive name.
```suggestion
  async getJobInformation(): Promise<{ job_id: number; html_url: string }> {
```